.github/workflows/agw-workflow.yml

```diff
@@ -375,6 +376,13 @@ jobs:
           # Run a simple echo command to pull down the image. This makes it a bit more clear how much time is spent on building Magma and not pulling down the image.
           run: |
             echo "Pulled the devontainer image!"
+      - name: Setup Bazel Base Image
+        uses: addnab/docker-run-action@v2
+        with:
+          image: ${{ env.BAZEL_BASE_IMAGE }}
+          # Run a simple echo command to pull down the image. This makes it a bit more clear how much time is spent on building Magma and not pulling down the image.
+          run: |
+            echo "Pulled the bazel base image!"
             bazel # pull down bazel, if bazel download fails we can fail before we do all the lengthy work below
```

Comment:
Would we still want to be able to execute the parts on the devcontainer even when the bazel pull fails? I.e. only pull the image right before use? In that case we would see at least from one part of the pipeline, if it can successfully build. :man_shrugging:

Reply:
This seems to be a little inconsistent. Almost all jobs are executed even if the bazel binary download fails (`if: always()`). Even the bazel coverage job has `if: always()`. It should be consolidated in what case we want to do a partial upload of generated coverage - but this is getting out of scope for this task.